### PR TITLE
[ProgressV2] Make progress table more screen reader accessible

### PR DESCRIPTION
**Pull request by lfryemason**
Changing the structure of an expanded lesson column to a table in order to better accommodate screen reader accessibility.

* Use `<table>` for each expanded lesson column
* Add a caption that explains what is in the `<table>`
* Add a hidden student header to each row in the expanded lesson. This will not be shown, but will provide context for screen readers
* Each cell is appropriately associated with three or four headers (using the `headers=` attribute). The three headers are: the student, the lesson, the level and the parent level, if it is a sublevel.
* Make level cells appropriately tab navigable and highlight in a sensible way. Before this PR, empty level cells were 0px and would highlight the top left corner when focused.

* No visible UI changes should have been made

Uploading Screen Recording 2024-04-02 at 10.45.21 AM.mov…

## Links
https://codedotorg.atlassian.net/browse/TEACH-882

## Testing story
Manually testing no UI changes on safari, firefox and chrome.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/sectionProgressV2/ExpandedProgressColumnHeader.jsx [hunk 1]
```diff
@@ -42,28 +43,32 @@ export default function ExpandedProgressColumnHeader({
...
-        <LessonTitleTooltip lesson={lesson} />
-        <FontAwesome
-          icon="caret-down"
-          className={styles.expandedHeaderCaret}
-          title={i18n.unexpand()}
-        />
-        <div className={styles.expandedHeaderLessonText}>{headerText}</div>
-      </div>
-      <div
+    <tbody className={styles.expandedHeader} key={lesson.id}>
```
Comment (hannahbergam) [inline apps/src/templates/sectionProgressV2/ExpandedProgressColumnHeader.jsx:46]: 🥹

File: apps/src/templates/sectionProgressV2/ExpandedProgressDataColumn.jsx [hunk 2]
```diff
@@ -39,64 +43,76 @@ function ExpandedProgressDataColumn({
...
-            return getExpandedChoiceLevel(level);
-          }
-          return [getSingleLevelColumn(level)];
-        })}
-      </div>
+      <tbody className={styles.expandedTable}>
+        {sortedStudents.map(student => (
+          <>
+            <th hidden={true} id={getStudentRowHeaderId(student.id)}>
+              {getFullName(student)}
```
Comment (hannahbergam) [inline apps/src/templates/sectionProgressV2/ExpandedProgressDataColumn.jsx:79]: I was hoping to see some sort of name retrieval after reading about the id- nice!

**Review by hannahbergam — APPROVED** [review]
SO EXCITING!! Excellent work. My only request would be a little screen recording of some tabNav, but non-blocking!

**Comment by lfryemason** [comment]
> SO EXCITING!! Excellent work. My only request would be a little screen recording of some tabNav, but non-blocking!

Added a video to the pr description

**Review by kobryan0619 — APPROVED** [review]
Really well thought out - I did a little check on Safari and Chrome and didn't see any visual differences either.  Nice work!

**Pull request merged**